### ISSUE TITLE
Supervised training

### ISSUE THREAD

**Issue by rnyak** [facebookresearch/fastText#995]
Thanks @Celebio for your quick response. "unsupervised training tutorial" does not respond all  my questions. 

I'd like to still ask the following questions:

When we do supervised training, apparently we use pre-learned word representations, right? How were the words representations learned? From Wikipedia's articles? with cbow or skipgram?

What's the total number of parameters of text classification model? Is there a way to print out the total number of params of the model?

**Comment by Celebio** [comment]
Hi @rnyak ,

> When we do supervised training, apparently we use pre-learned word representations, right?

Not necessarily. You can train your model with supervised learning without pretrained vectors: in this case, it will automatically compute word representations that are adapted to the classification task. You can refer to [this answer](https://github.com/facebookresearch/fastText/issues/469#issuecomment-486229924) for more information on that. If you want to force the word vectors learnt from an unsupervised training, then you can use `-pretrainedVectors` options.


> How were the words representations learned? From Wikipedia's articles? with cbow or skipgram?

The ones we propose to download on our website are from [common crawl](https://fasttext.cc/docs/en/crawl-vectors.html) and [wikipedia](https://fasttext.cc/docs/en/pretrained-vectors.html).

In order to see how a model was trained, you can use the `dump` command like this:

```
./fasttext dump mymodel.bin args
```
In particular, this will tell you if it was trained with cbow or skipgram.

If you want to know the size of your matrices, you can do:
```
./fasttext dump mymodel.bin input | head -n 1
./fasttext dump mymodel.bin output | head -n 1
```

Best regards,
Onur

**Comment by rnyak** [comment]
Thanks Onur.

**Comment by lucacerone** [comment]
Many thanks Onur (@Celebio)!

I would just like a couple of extra clarifications if possible!

- if I use pretrainedVectors to do supervised classification, would the embeddings be "fine tuned" before doing the classification? or they are just used as is and only the parameters for softmax (or hierarchical softmax) layer is trained?

- Facebook nicely makes available word embeddings for many languages, that use 5-chargrams. What if I want to use them to build a classification model that uses chargrams too, I am limited to use only 5 as an option, am I correct?

- in the "pretrained" embedding documentation, it is shown how to create embedding representations on out of vocabulary (OOV) words. It is my understanding that we can "append" the oov embeddings to an existing .vec file, and then use the "augmented" .vec file, as pretrainedVectors for classification purposes, is that correct? will the new oow embeddings be generated at "subword" level in the .vec file? (I assume that new chargrams might appear)

- if the above is correct, one could also concatenate two .vec files from different languages is that correct? In this case what happens if there are duplicates (e.g. false-friends, like the word )?

Many thanks for any clarification you might provide!

**Comment by indigoviolet** [comment]
@Celebio @lucacerone if you have answers to any of the questions above, it would be much appreciated even here at the end of 2022. My context is that I'm trying to use the aligned datasets to get word vectors, but they don't have subword info so I can't handle out-of-vocabulary words.
